### ISSUE TITLE
how did you create .env?

### ISSUE THREAD

**Issue by MatheusRV** [JamborJan/paperwork#49]
i'm porting new apps into snadstorm and i'm confused how to create .env

could you help?

**Comment by JamborJan** [comment]
Hey @MatheusRV, I don't get the point of question, sorry. I'm trying to help you as good as I can. Can you rephrase your question please?

Some random notes about my approach:
- I'm using the default Vagrant aproach and the Debian box provided by Sandstorms Vagrant-SPK: https://github.com/sandstorm-io/vagrant-spk
- Within [setup.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/setup.sh) you setup the debian box just like you would do it with a normal virtual machine

Please point me tou your repositories and I can have a look to help you.

**Comment by JamborJan** [comment]
When you say `.env` I guess you are talking about the Docker config file. Within vagrant this is done in the Vagrant file. My example: https://github.com/JamborJan/paperwork/blob/master/.sandstorm/Vagrantfile 

The equivalents of the Docker file in Vagrant are the [setup.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/setup.sh), [build.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/build.sh) and [launcher.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/launcher.sh)

Does that help you?

**Comment by MatheusRV** [comment]
Hi, tks for an answer. My English is terrible, I need do something about that.

I'm talking about .env of Laravel, I'm creating GitHub of the project right now.

**Comment by JamborJan** [comment]
Okay, I still don't get it so I try to explain how I did it with Paperwork. From my point of view it doesn't make sense to port laravel alone as it is a php framework and no application. Thats why I have asked about the actual application you are going to port.

1. I'm setting up a debian box with Vagrant 
2. All details which packages are installed can be found in [setup.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/setup.sh)
3. The actual laravel configuration is included in the master paperwork repository
4. I'm cloning the repository code as last step of the setup.sh
5. in [build.sh](https://github.com/JamborJan/paperwork/blob/master/.sandstorm/build.sh) I'm running `composer` which uses the provided composer.json from the upstream repository and installs laravel

I hope this somehow makes sense to you, if not please don't hesitate to ask again.

**Comment by MatheusRV** [comment]
I'm working on https://github.com/MatheusRV/Cachet-Sandstorm but the MYSQL is showing a strange error with socket then I go to vagrant-spj vm ssh and restart the MYSQL it works ok... Then I try to vagrant-spk dev I receive a Symlink error, reading on the internet I see is related to VirtualBox and Vagrant.

I'm current using Windows Linux Subsystem, Vagrant 1.9.5 that support VirtualBox over windows and vagrant on WSL.

Tks

**Comment by JamborJan** [comment]
I have checked your code, this part:

```
sudo /etc/init.d/mysql stop
sudo /etc/init.d/mysql start
echo "CREATE DATABASE IF NOT EXISTS cachet; GRANT ALL on cachet.* TO 'cachet'@'localhost' IDENTIFIED BY 'cachet'; FLUSH PRIVILEGES;" | mysql -h localhost -u root
php artisan key:generate
php artisan app:install
```

Should be deleted in build.sh and copied to launcher.sh then mysql should work. *BUT* I highly recommend to use SQLite instead of mysql because it has better performance within Sandstorm. Furthermore one very important thing: all folders which must be writable must be located wuthin /var/. I never used Cachet, do you know which folders must be writable? If yes please let me know and I can create a Pull Request with all changes to make this work.

**Comment by MatheusRV** [comment]
Just ```
chmod 777 /opt/app/database/database.sqlite
chmod -R 777 /opt/app/bootstrap/
chmod -R 777 /opt/app/storage
``` but this command does not work...

i change for sqlite and works well.. tks

**Comment by JamborJan** [comment]
Ok, will close this one here.
